drivers/display/display_mcux_elcdif.c

```diff
@@ -227,47 +257,43 @@ static const struct display_driver_api mcux_elcdif_api = {
 	.set_orientation = mcux_elcdif_set_orientation,
 };
 
-static void mcux_elcdif_config_func_1(const struct device *dev);
-
-static struct mcux_elcdif_config mcux_elcdif_config_1 = {
-	.base = (LCDIF_Type *) DT_INST_REG_ADDR(0),
-	.irq_config_func = mcux_elcdif_config_func_1,
-#ifdef CONFIG_MCUX_ELCDIF_PANEL_RK043FN02H
-	.rgb_mode = {
-		.panelWidth = 480,
-		.panelHeight = 272,
-		.hsw = 41,
-		.hfp = 4,
-		.hbp = 8,
-		.vsw = 10,
-		.vfp = 4,
-		.vbp = 2,
-		.polarityFlags = kELCDIF_DataEnableActiveHigh |
-				 kELCDIF_VsyncActiveLow |
-				 kELCDIF_HsyncActiveLow |
-				 kELCDIF_DriveDataOnRisingClkEdge,
-		.pixelFormat = kELCDIF_PixelFormatRGB565,
-		.dataBus = kELCDIF_DataBus16Bit,
-	},
-	.pixel_format = PIXEL_FORMAT_BGR_565,
-	.bits_per_pixel = 16,
-#endif
-};
-
-static struct mcux_elcdif_data mcux_elcdif_data_1;
-
-DEVICE_DT_INST_DEFINE(0,
-		    &mcux_elcdif_init,
-		    NULL,
-		    &mcux_elcdif_data_1, &mcux_elcdif_config_1,
-		    POST_KERNEL, CONFIG_DISPLAY_INIT_PRIORITY,
-		    &mcux_elcdif_api);
-
-static void mcux_elcdif_config_func_1(const struct device *dev)
-{
-	IRQ_CONNECT(DT_INST_IRQN(0),
-		    DT_INST_IRQ(0, priority),
-		    mcux_elcdif_isr, DEVICE_DT_INST_GET(0), 0);
+#define MCUX_ELDCIF_DEVICE(id)							\
+	static void mcux_elcdif_config_func_##id(const struct device *dev);	\
+	static const struct mcux_elcdif_config mcux_elcdif_config_##id = {	\
+		.base = (LCDIF_Type *) DT_INST_REG_ADDR(id),			\
+		.irq_config_func = mcux_elcdif_config_func_##id,		\
+		.rgb_mode = {							\
+			.panelWidth = DT_INST_PROP(id, width),			\
+			.panelHeight = DT_INST_PROP(id, height),		\
+			.hsw = DT_INST_PROP(id, hsync),				\
+			.hfp = DT_INST_PROP(id, hfp),				\
+			.hbp = DT_INST_PROP(id, hbp),				\
+			.vsw = DT_INST_PROP(id, vsync),				\
+			.vfp = DT_INST_PROP(id, vfp),				\
+			.vbp = DT_INST_PROP(id, vbp),				\
+			.polarityFlags = DT_INST_PROP(id, polarity),		\
+			.dataBus = LCDIF_CTRL_LCD_DATABUS_WIDTH(		\
+					DT_INST_ENUM_IDX(id, data_buswidth)),	\
+		},								\
+		.pixel_format = DT_INST_ENUM_IDX(id, pixel_format),		\
```

Comment:
Thanks for adding the units. Looking at the uses in the driver, these values are being read unconditionally, so it also looks like the binding should make them `required: true`.

Reply:
Thank you for your comment, I have update the dts file.

Reply:
@mmahadevan108 I don't understand your response, sorry. My request was related to the binding, not the DTS file:

> the binding should make them required: true.

I don't see this change

Reply:
I have added a default value instead, is'nt that similar to making them required?

Reply:
Oh, sorry, I missed that. But now this is breaking the rules about using default without a justification in the description, so that needs to be fixed, please.

Reply:
Thanks, updated.